src/dialects/postgres/index.js

```diff
@@ -266,6 +267,30 @@ assign(Client_PG.prototype, {
     }
     return resp;
   },
+
+  canCancelQuery: true,
+  cancelQuery(connectionToKill) {
+    const acquiringConn = this.acquireConnection();
+
+    // Error out if we can't acquire connection in time.
+    // Purposely not putting timeout on `pg_terminate_backend` execution because erroring
+    // early there would release the `connectionToKill` back to the pool with
+    // a `KILL QUERY` command yet to finish.
+    return acquiringConn
+      .timeout(100)
```

Comment:
This special timeout here seems dangerous to me. 

That timeout it is done by Bluebird instead of pool I suppose. So when it triggers the promise chain rejects and finally connection is not released. 

But since timeout was done by Bluebird, the `this.acquireConnection()` method call will not know that it was timed out and it keeps running and might get connection in the end. Anyhow now the original promise chain was already rejected, so there is no more handler for that connection which was got and it is just ignored and never returned to pool.

So shouldn't we just trust the pool's acquireConnectionTimeout to trigger here instead of using bluebirds timeout() ?

Reply:
You're correct. Will fix.
Also found that the test for this was not even being triggered, fixing that as well.